### PR TITLE
actions: Upload test results to opensearch service

### PR DESCRIPTION
**Pull request by nashif**
On push, upload test results to opensearch for analysis and reporting.
Goal is to use this data to understand test coverage better and use this
services for all test reporting, also for results coming from testing on
hardware.
Opensearch is currenly being used for evaluation, we are considering
the switch to elasticsearch later.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: .github/workflows/twister.yaml [hunk 1]
```diff
@@ -259,22 +259,41 @@ jobs:
...
       # the build-and-test job might be skipped, we don't need to run this job then
     if: success() || failure()
 
     steps:
+      # Needed for opensearch and upload script
+      - if: github.event_name == 'push'
+        name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (stephanosio) [inline .github/workflows/twister.yaml:282]: should not use `github.event.pull_request.head.sha` for push events. `ref` here can be omitted.

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by stephanosio — APPROVED** [review]

File: scripts/ci/upload_test_results.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    client = OpenSearch(
+            hosts = [{'host': host, 'port': port}],
+            http_auth=auth,
+            use_ssl=True,
+            verify_certs = False,
+            ssl_assert_hostname = False,
+            ssl_show_warn = False,
+    )
+    # Create an index with non-default settings.
+    index_name = 'zephyr-main-2'
```
Comment (galak) [inline scripts/ci/upload_test_results.py:31]: any reason for this particular name?

**Comment by henrikbrixandersen** [comment]
Will the community (eventually) have access to the uploaded, searchable data?

**Comment by stephanosio** [comment]
> Will the community (eventually) have access to the uploaded, searchable data?

Eventually, yes. The idea is to use OpenSearch + Dashboard (or ElasticSearch + Kibana for that matter) like a test management system.

**Review by galak — APPROVED** [review]
In general looks good.  Will want to cleanup the upload script at some point.

**Comment by nashif** [comment]
> Will the community (eventually) have access to the uploaded, searchable data?

thats the idea.

**Comment by nashif** [comment]
@henrikbrixandersen a sneak preview:

image

**Pull request merged**